File: LeetCode Challenge/LeetCode Youtube/YuhLapTrinhVien/Vol(0-199)/1523/1523.go
package main

import "fmt"

// Code in video: O(n) -> O(1)
func countOdds(low int, high int) int {
	result := 0
	i := low
	oddNumbers := []int{} // Slice to store odd numbers

	for i <= high {
		if i%2 == 1 {
			fmt.Println("Found Odd:", i)
			oddNumbers = append(oddNumbers, i) // Store the odd number
			result++
		}
		i++
	}

	fmt.Println("Odd numbers in range:", oddNumbers)
	return result
}

func main() {
	oddArray := [2]int{3, 7}
	fmt.Println("Total Odd Numbers:", countOdds(oddArray[0], oddArray[1]))
}

File: LeetCode Challenge/LeetCode Youtube/go-leetcode/YuhLapTrinhVien/Vol(0-199)/20-Question/1523/1523.ts
function countOdds(low: number, high: number): number {
  let result = 0;
  let i = low;
  let oddNumbers: number[] = []; // Array to store odd numbers

  while (i <= high) {
    if (i % 2 === 1) {
      console.log("Found Odd:", i);
      oddNumbers.push(i); // Store the odd number
      result++;
    }
    i++;
  }

  console.log("Odd numbers in range:", oddNumbers);
  return result;
}

function countOdds_1(low: number, high: number): number {
  let numOfElement = high - low + 1;
  let result = Math.floor(numOfElement / 2);

  if (numOfElement % 2 === 1 && low % 2 === 1) {
    result++;
  }
  return result;
}

// Test case
const oddArray: [number, number] = [3, 7];

console.log("Total Odd Numbers 1:", countOdds(oddArray[0], oddArray[1]));

console.log("------------------------");

console.log("Total Odd Numbers 2:", countOdds_1(oddArray[0], oddArray[1]));
